### PR TITLE
Refactor error handling to display more user friendly errors

### PR DESCRIPTION
**Pull request by shubhamkmr04**
# Description

Relates to issue: **[ZEUS-2003](https://github.com/ZeusLN/zeus/issues/2003)**

This pull request is categorized as a:

- [ ] New feature
- [X] Bug fix
- [X] Code refactor
- [ ] Configuration change
- [ ] Locales update
- [ ] Quality assurance
- [ ] Other

## Checklist
- [X] I’ve run `yarn run tsc` and made sure my code compiles correctly
- [X] I’ve run `yarn run lint` and made sure my code didn’t contain any problematic patterns
- [X] I’ve run `yarn run prettier` and made sure my code is formatted correctly
- [X] I’ve run `yarn run test` and made sure all of the tests pass

## Testing

If you modified or added a utility file, did you add new unit tests?

- [ ] No, I’m a fool
- [X] Yes
- [ ] N/A

I have tested this PR on the following platforms (please specify OS version and phone model/VM):

- [X] Android
- [X] iOS

I have tested this PR with the following types of nodes (please specify node version and API version where appropriate):

- [ ] Embedded LND
- [X] LND (REST)
- [ ] LND (Lightning Node Connect)
- [ ] Core Lightning (c-lightning-REST)
- [ ] LndHub
- [ ] [DEPRECATED] Core Lightning (Spark)
- [ ] [DEPRECATED] Eclair

### Locales
- [ ] I’ve added new locale text that requires translations
- [ ] I’m aware that new translations should be made on the ZEUS [Transfix page](https://app.transifex.com/ZeusLN/zeus/) and not directly to this repo

### Third Party Dependencies and Packages

- [ ] Contributors will need to run `yarn` after this PR is merged in
- [ ] 3rd party dependencies have been modified:
    * verify that `package.json` and `yarn.lock` have been properly updated
    * verify that dependencies are installed for both iOS and Android platforms

### Other:

- [ ] Changes were made that require an update to the README
- [ ] Changes were made that require an update to onboarding

### CODE REVIEW

File: utils/ErrorUtils.ts [hunk 1]
```diff
@@ -15,6 +15,23 @@ const userFriendlyErrors: any = {
 };
 
 const errorToUserFriendly = (error: string, localize = true) => {
```
Comment (kaloudis) [inline utils/ErrorUtils.ts:17]: love this approach of extending this function

File: utils/ErrorUtils.test.ts [hunk 2]
```diff
@@ -3,6 +3,28 @@ import { errorToUserFriendly } from './ErrorUtils';
...
+                    JSON.stringify({
+                        code: 3,
+                        message:
+                            'proto: (line 1:126): invalid value for uint64 type: "0.01"',
+                        details: []
+                    })
+                )
+            ).toEqual(
+                'proto: (line 1:126): invalid value for uint64 type: "0.01"'
+            );
```
Comment (kaloudis) [inline utils/ErrorUtils.test.ts:27]: good tests here but you should add cases where message is on `errorObject.error.message`

File: stores/ChannelsStore.ts [hunk 3]
```diff
@@ -510,12 +511,16 @@ export default class ChannelsStore {
                         if (!connectPeerOnly) {
                             this.channelRequest = request;
                         } else {
-                            this.errorMsgPeer = error.toString();
+                            this.errorMsgPeer = errorToUserFriendly(
+                                error.message.toString()
```
Comment (kaloudis) [inline stores/ChannelsStore.ts:515]: why do we have to pick message off the object and convert to string here?

We should be able to pass in the error raw and have the function handle the rest, ie. `errorToUserFriendly(error)`

**Review by kaloudis — CHANGES_REQUESTED** [review]

File: stores/ChannelsStore.ts (hunk 3 above)
Comment (shubhamkmr04) [reply stores/ChannelsStore.ts]: Made the changes accordingly.

File: utils/ErrorUtils.ts [hunk 4]
```diff
@@ -14,16 +14,38 @@ const userFriendlyErrors: any = {
...
 };
 
-const errorToUserFriendly = (error: string, localize = true) => {
-    if (localize) {
-        const localeString = require('./LocaleUtils').localeString;
-        return (
-            localeString(userFriendlyErrors[error])?.replace('Zeus', 'ZEUS') ||
-            error
-        );
+const errorToUserFriendly = (error: any, localize = true) => {
```
Comment (kaloudis) [inline utils/ErrorUtils.ts:17]: this is still incorrect. Type should be of `:Error` not `:any`

File: stores/ChannelsStore.ts [hunk 5]
```diff
@@ -541,7 +542,7 @@ export default class ChannelsStore {
                 this.channelSuccess = true;
             })
             .catch((error: any) => {
```
Comment (kaloudis) [inline stores/ChannelsStore.ts:544]: change the type here and on line 501 to `Error`

File: stores/InvoicesStore.ts [hunk 6]
```diff
@@ -402,7 +403,7 @@ export default class InvoicesStore {
             .catch((error: any) => {
```
Comment (kaloudis) [inline stores/InvoicesStore.ts:403]: change the type here to `Error`

File: utils/ErrorUtils.ts [hunk 7]
```diff
@@ -14,16 +14,38 @@ const userFriendlyErrors: any = {
...
 
-const errorToUserFriendly = (error: string, localize = true) => {
-    if (localize) {
-        const localeString = require('./LocaleUtils').localeString;
-        return (
-            localeString(userFriendlyErrors[error])?.replace('Zeus', 'ZEUS') ||
-            error
-        );
+const errorToUserFriendly = (error: Error, localize = true) => {
+    if (typeof error === 'object') {
```
Comment (kaloudis) [inline utils/ErrorUtils.ts:18]: shouldn't type always be `Error` now?

File: utils/ErrorUtils.ts [hunk 8]
```diff
@@ -14,16 +14,38 @@ const userFriendlyErrors: any = {
...
+            return errorMessage?.message;
+        }
+
+        if (
+            errorMessage &&
+            errorMessage?.error &&
+            errorMessage?.error?.message
+        ) {
+            return errorMessage?.error?.message;
+        }
```
Comment (kaloudis) [inline utils/ErrorUtils.ts:35]: I don't think this block should be returning the message as is. We should probably try to see if we can localize it after we parse the error string out of the object

File: utils/ErrorUtils.test.ts [hunk 9]
```diff
@@ -3,6 +3,38 @@ import { errorToUserFriendly } from './ErrorUtils';
...
     describe('errorToUserFriendly', () => {
         it('Turns error message to user friendly values', () => {
+            expect(
+                errorToUserFriendly({
+                    message: `{
+                        "code": 2,
+                        "message": "transaction output is dust",
+                        "details": []
+                    }`
+                })
```
Comment (kaloudis) [inline utils/ErrorUtils.test.ts:13]: I thought we would have had to wrap the inputs w/ Error constructors here. Interesting that it all still passes.

File: utils/ErrorUtils.ts [hunk 10]
```diff
@@ -14,16 +14,33 @@ const userFriendlyErrors: any = {
         'error.failureReasonInsufficientBalance'
 };
 
-const errorToUserFriendly = (error: string, localize = true) => {
+const errorToUserFriendly = (error: Error, localize = true) => {
+    let errorMessage: object | string | any = error.message;
```
Comment (kaloudis) [inline utils/ErrorUtils.ts:18]: this is getting much closer but not quite, right.

Shouldn't this always be a `string`? The case in which it's an object is not covered in the test cases. The test cases should also include actual `Error` objects being passed in

Comment (shubhamkmr04) [reply utils/ErrorUtils.ts]: Yes, you are right, error.message is always string  in all cases. Also added `Error` objects in test cases in the recent commit

**Review by kaloudis — APPROVED** [review]
LGTM

**Pull request merged**